File: python3/set_mismatch.py
"""
Space   : O(1)
Time    : O(n)
"""


class Solution:
    def findErrorNums(self, nums: List[int]) -> List[int]:
        n = len(nums)
        total_range, total_nums = sum([i for i in range(1, n+1)]), sum(nums)

        double = total_nums - sum(set(nums))

        return [double, total_range - total_nums + double]

File: golang/set_mismatch.go
package main

// Space   : O(1)
// Time    : O(n)

func findErrorNums(nums []int) []int {
	n := len(nums)
	dict := make(map[int]int)
	totalVal, totalIdx, totalUnique := 0, 0, 0

	for i := 0; i < n; i++ {
		if _, ok := dict[nums[i]]; !ok {
			dict[nums[i]] = 1
		}
		totalVal += nums[i]
		totalIdx += i + 1
	}

	for k := range dict {
		totalUnique += k
	}

	return []int{totalVal - totalUnique, totalIdx - totalUnique}
}
